sound/soc/sof/ipc4-topology.c

```diff
@@ -1957,6 +1957,47 @@ static void sof_ipc4_put_queue_id(struct snd_sof_widget *swidget, int queue_id,
 	ida_free(queue_ida, queue_id);
 }
 
+static int sof_ipc4_set_copier_sink_format(struct snd_sof_dev *sdev,
+					   struct snd_sof_widget *src_widget,
+					   struct snd_sof_widget *sink_widget,
+					   int sink_id)
+{
+	struct sof_ipc4_base_module_cfg *src_config = src_widget->private;
+	struct sof_ipc4_base_module_cfg *sink_config = sink_widget->private;
+	struct sof_ipc4_copier_config_set_sink_format format;
+	struct sof_ipc4_fw_module *fw_module;
+	struct sof_ipc4_msg msg = {{ 0 }};
+	u32 header, extension;
+
+	dev_dbg(sdev->dev, "%s set copier sink %d format\n",
```

Comment:
@bardliao can you elaborate a bit in your commit message? I unerstand this is a requirement but why is it needed would be helpful. So what happens when a copier is connected to sink Id 0, where does it get the format from?

Reply:
> @bardliao can you elaborate a bit in your commit message? I unerstand this is a requirement but why is it needed would be helpful. So what happens when a copier is connected to sink Id 0, where does it get the format from?

Yes, done

Reply:
@bardliao you only partly answered to @ranj063 's question, where the src_id zero case handled